pkg/artifacts/signable.go

```diff
@@ -257,6 +276,59 @@ func extractTargetFromResults(obj objects.TektonObject, identifierSuffix string,
 	return ss
 }
 
+// ExtractStructuredTargetFromResults extracts structured signable targets aim to generate intoto provenance as materials within TaskRun results and store them as StructuredSignable.
+func ExtractStructuredTargetFromResults(obj objects.TektonObject, categoryMarker string, logger *zap.SugaredLogger) []*StructuredSignable {
+	objs := []*StructuredSignable{}
+	for _, res := range obj.GetResults() {
+		if strings.HasSuffix(res.Name, categoryMarker) {
+			if validateObjectKeys(structuredSignableSchema.Properties, &res.Value) != nil {
+				logger.Errorf("%s should follow this schema: %v", res.Name, structuredSignableSchema)
```

Comment:
Drop this and other log message here to warn - this is a problem with what the user gave us, not with how chains is processing this.

Reply:
Can you explain a bit more on this? I'm trying to understand that maybe we are having this strategy that when users are not passing the right schema, Chains should just ignore it. But when it has the type hinting marker, should we expect the schema should be right and error / warn about it?